torchvision/transforms/transforms.py

```diff
@@ -753,21 +757,39 @@ def get_params(brightness, contrast, saturation, hue):
             Transform which randomly adjusts brightness, contrast and
             saturation in a random order.
         """
+        def _sample_from(value, name):
+            factor = None
+            if isinstance(value, numbers.Number) and value >= 0:
+                if name == 'hue':
+                    factor = random.uniform(-value, value)
```

Comment:
I'd prefer not to special-case for the `hue` value. What about something like the following:

```python
def _sample_from(value, name, center=1, clip_first_on_zero=True):
    if isinstance(value, numbers.Number) and value >= 0:
        value = [center - value, center + value]
    if clip_first_on_zero:
        value[0] = max(value[0], 0)
    factor = random.uniform(value[0], value[1])
```

Reply:
when the input value is invalid, for example `brightness=(-0.5, 0.5)`, should we raise an error or auto clip it?

Reply:
the current code doesn't raise a warning, right?
Or is it handled in the underlying `F.adjust_brightness`? I think the error should be there, right?

Reply:
Only `F.adjust_hue` checks the input, brightness/contrast/saturation don't.  
The current code only need to ensure `brightness > 0`, but what if the input is `(0.5, -0.5)`, which means `min > max`, so I think we should check here.

Reply:
Or should we check the input in `__init__`, just as `Line 911`(https://github.com/yaox12/vision/blob/master/torchvision/transforms/transforms.py#L911)

Reply:
Let's check it in `__init__`, might be better